packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -2508,6 +2508,19 @@ const getCommonWorkflowFunctions = (
 	prepareOutputData: async (outputData) => [outputData],
 });
 
+const executionCancellationFunctions = (
+	abortSignal: AbortSignal,
+): Pick<IExecuteFunctions, 'onExecutionCancellation' | 'getExecutionCancelSignal'> => ({
+	getExecutionCancelSignal: () => abortSignal,
+	onExecutionCancellation: (handler) => {
```

Comment:
Would it make sense to unit test this?

Reply:
not sure. I generally prefer to unit tests stuff that has logic in it. not sure if it make sense to unit test if the event listeners is correctly attached. maybe it makes sense to test that the handler is invoked only once, but I think that can happen later.